src/main/java/com/ibm/watson/modelmesh/payload/PayloadDataProcessor.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2023 IBM Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy
+ * of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.ibm.watson.modelmesh.payload;
+
+import java.util.function.Function;
+
+import io.netty.buffer.ByteBuf;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * {@link PayloadProcessor} abstract implementation for {@link PayloadProcessor}s that need to access {@link ByteBuf}s
+ * contained in {@link Payload}s.
+ * This class takes care of retaining and releasing such {@link ByteBuf}s during payload processing.
+ * Extensions of this class should not modify {@link ByteBuf}s contained in {@link Payload}s.
+ */
+public abstract class PayloadDataProcessor implements PayloadProcessor{
```

Comment:
I'm not sure why this class is needed. The async one can handle the ref counting.

Reply:
ok, will drop this.